packages/personal-data-erasure/src/main.js

```diff
@@ -106,92 +134,111 @@ export default class PersonalDataErasure {
+      },
+      { accumulate: true }
+    )
+    return flatten(
+      messages.map(
+        (response: ClientResponse): Messages | void =>
+          response.body ? response.body.results : undefined
+      )
+    )
   }
 
-  deleteAll(customerId) {
+  deleteAll(customerId: string): Promise<AllData> {
     if (!customerId) throw Error('missing `customerId` argument')
     this.logger.info('Starting deletion')
 
     const requestBuilder = createRequestBuilder({
       projectKey: this.apiConfig.projectKey,
     })
-    const customersUri = {
+    const customersResource = {
       uri: requestBuilder.customers.where(`id = "${customerId}"`).build(),
       builder: requestBuilder.customers,
     }
-    const ordersUri = {
+    const ordersResource = {
       uri: requestBuilder.orders.where(`customerId = "${customerId}"`).build(),
       builder: requestBuilder.orders,
     }
-    const cartsUri = {
+    const cartsResource = {
       uri: requestBuilder.carts.where(`customerId = "${customerId}"`).build(),
       builder: requestBuilder.carts,
     }
-    const paymentsUri = {
+    const paymentsResource = {
       uri: requestBuilder.payments
         .where(`customer(id = "${customerId}")`)
         .build(),
       builder: requestBuilder.payments,
     }
-    const shoppingsListsUri = {
+    const shoppingListsResource = {
       uri: requestBuilder.shoppingLists
         .where(`customer(id = "${customerId}")`)
         .build(),
       builder: requestBuilder.shoppingLists,
     }
-    const reviewsUri = {
+    const reviewsResource = {
       uri: requestBuilder.reviews
         .where(`customer(id = "${customerId}")`)
         .build(),
       builder: requestBuilder.reviews,
     }
 
-    const urisOfResourcesToDelete = [
-      customersUri,
-      ordersUri,
-      cartsUri,
-      paymentsUri,
-      shoppingsListsUri,
-      reviewsUri,
+    const resourcesToDelete = [
+      customersResource,
+      ordersResource,
+      cartsResource,
+      paymentsResource,
+      shoppingListsResource,
+      reviewsResource,
     ]
 
     return Promise.all(
-      urisOfResourcesToDelete.map(uri => {
-        const request = PersonalDataErasure.buildRequest(uri.uri, 'GET')
-
-        return this.client.process(request, response => {
-          if (response.statusCode !== 200 && response.statusCode !== 404)
-            return Promise.reject(
-              Error(`Request returned status code ${response.statusCode}`)
-            )
-          if (response.statusCode === 200)
-            this._deleteOne(response, uri.builder)
-          return Promise.resolve()
-        })
-      })
+      resourcesToDelete.map(
+        (resource: {
+          uri: string,
+          builder: ServiceBuilderInstance,
+        }): Promise<any> => {
+          const request = PersonalDataErasure.buildRequest(resource.uri, 'GET')
+
+          return this.client.process(
+            request,
+            (response: Object): Promise<any> => {
```

Comment:
Isn’t this `Promise<AllData>`?

Reply:
Im afraid that doesn't work as we resolve the promise without data like this.
`return Promise.resolve()`

Setting it to `Promise<AllData>` then returns this error instead

```js
Cannot return `Promise.resolve()` because undefined [1] is incompatible with `AllData` [2] in type argument `R` [3].

   packages/personal-data-erasure/src/main.js:227:22
   227|               return Promise.resolve()
                             ^^^^^^^^^^^^^^^^^ [1]

References:
   packages/personal-data-erasure/src/main.js:220:41
   220|             (response: Object): Promise<AllData> => {
                                                ^^^^^^^ [2]
   /private/tmp/flow/flowlib_3e57fb1b/core.js:605:24
   605| declare class Promise<+R> {
                               ^ [3]
```

Reply:
Then it should just be `Promise<void>` 😉